src/Components/Admin/Rooms/roomModal.js

```diff
@@ -50,24 +58,61 @@ export default function RoomModal({
         setFriendlyName(event.target.value);
     };
 
+    const handleRoomSolveTimeChange = (event) => {
+        setRoomSolveTime(event.target.value);
+    };
+
     const handleModalCloseClick = () => {
         handleModalClose();
         setRoomName("");
         setRoomDescription("");
         setFriendlyName("");
+        setRoomSolveTime("");
+        setIsPublished(false);
+        setIsPreviewable(false);
     };
 
     const handleModalSubmitClick = () => {
-        handleSubmit({
-            name: roomName,
-            description: roomDescription,
-            friendly_name: friendlyName,
-        });
+        if (isEdit) {
+            handleSubmit({
+                name: roomName,
+                description: roomDescription,
+                friendly_name: friendlyName,
+                is_published: isPublished,
+                is_previewable: isPreviewable,
+                expected_solve_time: roomSolveTime,
+            });
+            setRoomSolveTime("");
+            setIsPublished(false);
+            setIsPreviewable(false);
+        } else {
+            handleSubmit({
+                name: roomName,
+                description: roomDescription,
+                friendly_name: friendlyName,
+            });
+        }
         setRoomName("");
         setRoomDescription("");
         setFriendlyName("");
     };
 
+    const handleIsPublishedClick = () => {
+        if (isPublished === true) {
+            setIsPublished(false);
+        } else {
+            setIsPublished(true);
+        }
```

Comment:
Can be simplified to:
```
setIsPublished(!isPublished)
```
😄

Reply:
I knew there must have been a smarter way to do that. Thank you for your wisdom!